.circleci/config.yml

```diff
@@ -0,0 +1,46 @@
+version: 2
+# 2.1 does not yet support local run
+# unless with workaround. For simplicity just use it.
+# https://github.com/CircleCI-Public/circleci-cli/issues/79
+
+jobs:
+  build_and_test:
+    docker:
+      - image: circleci/node:8
+    steps:
+      - checkout
+      # Download and cache dependencies
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ checksum "package-lock.json" }}
+            # fallback to using the latest cache if no exact match is found
+            - v1-dependencies-
+      - run:
+          name: Install npm from lock
+          command: npm ci
```

Comment:
If we're caching dependencies I believe we should be using `npm install`, because `npm ci` deletes the node_modules directory.

Reply:
Then that's a problem. It seems currently there is no way (to my best knowledge) to both use npm install but also rely on package-lock.json that allows caching. Context: https://github.com/npm/npm/issues/18286

Should we return to just use `npm install` for now?  At least this is the old Travis CI configuration behavior.

I hope to check-in this commit sooner because the longer it takes, the more merge resolutions are needed and potentially needs to change many things.

Reply:
But `npm install` does use the lockfile to decide what to install! Only when package.json asks for something incompatible is the lockfile ignored (for that particular package).

Agree on merging this PR as soon as the last discussed changes are ready.